Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -408,26 +408,27 @@ namespace GradientSignal
                 y = (height - 1) - y;
 
                 // For terrarium, there is a separate algorithm for retrieving the value
-                if (m_currentChannel == ChannelToUse::Terrarium)
-                {
-                    return GetTerrariumPixelValue(x, y);
-                }
-
-                const float value = AZ::RPI::GetImageDataPixelValue<float>(m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(m_currentChannel));
-                return (value * m_multiplier) + m_offset;
+                const float value = (m_currentChannel == ChannelToUse::Terrarium)
+                    ? GetTerrariumPixelValue(x, y, m_imageData)
+                    : AZ::RPI::GetImageDataPixelValue<float>(
+                        m_imageData, imageDescriptor, x, y, aznumeric_cast<AZ::u8>(m_currentChannel));
+
+                // Scale (inverse lerp) the value into a 0 - 1 range. We also clamp it because manual scale values could cause
+                // the result to fall outside of the expected output range.
+                return AZStd::clamp((value - m_offset) * m_multiplier, 0.0f, 1.0f);
             }
         }
 
         return defaultValue;
     }
 
-    float ImageGradientComponent::GetTerrariumPixelValue(AZ::u32 x, AZ::u32 y) const
+    float ImageGradientComponent::GetTerrariumPixelValue(AZ::u32 x, AZ::u32 y, AZStd::span<const uint8_t> imageData) const
```

Comment:
The reason I'm passing it in is because `ImageGradientComponent::SetupAutoScaleMultiplierAndOffset()` is calling this method as well now, and m_imageData isn't initialized at that point. I could switch around the initialization order, but that makes things a bit more fragile and is a little bit complicated because of the various optional checks and early-outs in the advanced settings. This seemed more straightforward and removes a bit of state dependency from that method.

Reply:
OK sounds good. I can cleanup fix-up to use the right mip channel as part of my PR.